### DISCUSSION TITLE
MCP server contribution

### DISCUSSION THREAD

**Discussion by BruceWouaigne** [modelcontextprotocol/modelcontextprotocol#1201] [Q&A - Server implementation] [answered]
Hi there! First, thanks to Anthropic for releasing this protocol. Here at [Linkup](https://linkup.so/), we share your vision for advancing the agentic web and see immense potential in the protocol to accelerate its development.

Now, I've got a question regarding contributions. As part of our commitment to this vision, we are in the process of implementing our own MCP server and wondering if we should use this repo or host it in our own GitHub org.

Is the goal of this repo to serve as a centralized hub for MCP server hosting, building, and publishing (for the thousands of MPC servers to come)?

Or should we, instead, take care of hosting, building and publishing and open PRs just to update the list of featured servers?

Thanks!

**Comment by dsp-ant** [comment]
We are currently thinking about how exactly we want to do this, particularly as servers become more complex and require more maintenance. We want to make sure there is no bottleneck in reviewing and releasing servers that a centrally maintained repository would have. I'd love to hear your opinion what you think is best and if there are any other options we haven't considered.

* We are considering adding a "community-servers" or restructuring the servers repo so it contains links to servers that are maintained outside of the modelcontextprotocol org. This will allow people to maintain servers by themselves at the pace they want and with the tools they want, while keeping a discoverable place for servers. This means we only have to maintain a somewhat up-to-date list.
* We could make servers the main repository for servers, but honestly, I don't think that scales well and greatly restricts agency of anybody without direct push permissions. it has the benefit of discoverability at the expense of severely limiting the agency of people to maintain their own servers well.
* The alternative to either of these options is to not do anything.

Generally, I think it is best to start severs outside of the servers repository. The servers repository is mostly intended as a good starting place for people who want to play around with servers.

**Reply by jspahrsummers** [reply, marked as answer]
We've now ratified some changes to how we'll manage the [servers](https://github.com/modelcontextprotocol/servers) repo going forward.

**TLDR:** We're restructuring the MCP [servers](https://github.com/modelcontextprotocol/servers) repository to focus on reference implementations only. New server implementations should be developed in separate repositories and can be added as links to our README, rather than submitted as PRs to this repository.

Please check out [the announcement](https://github.com/orgs/modelcontextprotocol/discussions/61) for full details!

**Comment by BruceWouaigne** [comment]
Thanks, David - I completely agree with your assessment. Having a single central repository for all servers won't scale. I like your suggestion of creating a separate community-servers repository that acts as a directory of links to other repos.
This reminds me of Langchain's journey, if I'm not mistaken, they started by hosting all integrations in one place but eventually transitioned to maintaining just an index of available integrations.

A two-repo solution would make a lot of sense to me:

1. A read-only repository with your reference implementations that shows developers what's possible
2. A separate repository focused solely on maintaining a list of available servers

This would give you the best of both worlds - you can showcase proven examples while making it easy for the community to discover and contribute their own implementations.

**Comment by zzh8829** [comment]
I asked Anthropic Claude on the best strategy moving forward here is the suggestions 
```
1. Move forward with creating a "community-servers" repository that would:
   - Serve as a directory/index of MCP server implementations
   - Contain links to externally maintained server repositories
   - Have clear contribution guidelines for adding new server listings
   - Include basic metadata and status information for each listed server

2. Keep the existing servers repository as:
   - A reference implementation space
   - A learning resource with well-documented examples
   - A starting point for developers new to MCP

3. Implement specific improvements:
   - Create clear documentation explaining the two-repository structure
   - Set up automated validation for the community server listings
   - Establish a straightforward PR process for adding new server entries
   - Add status badges or health checks for listed servers

This approach would:
- Scale better as the ecosystem grows
- Give server maintainers more autonomy
- Reduce bottlenecks in the review process
- Maintain discoverability through a central directory
- Follow successful patterns from other projects like Langchain
```

**Comment by zzh8829** [comment]
For context, [we](https://loglmhq.com/) have successfully integrated MCP into our product and also running multiple custom servers in production, having a place to contribute back would be greatly helpful to the community !

**Comment by sdairs** [comment]
Hey, in the interim I've created an unofficial repo to start tracking community MCP work (clients, servers and other projects) https://github.com/sdairs/mcp-community - please feel free to PR links to what you're working on!

**Comment by BruceWouaigne** [comment]
This nice registry also popped-up [https://mcp-get.com/](https://mcp-get.com/)

**Comment by localden** [comment]
We've limited Discussions in this repository to meeting notes and are closing this thread. To continue the conversation, pick a channel from [Contributor Communication](https://modelcontextprotocol.io/community/communication). Thanks for posting.
